src/dmadata/dmadata.c

```diff
@@ -0,0 +1,28 @@
+#include "segment_symbols.h"
+#include "z64dma.h"
+
+// Linker symbol declarations (used in the table below)
+#define DEFINE_DMA_ENTRY(name, _nameString) DECLARE_ROM_SEGMENT(name)
+#define DEFINE_DMA_ENTRY_UNSET(name, _nameString) DECLARE_ROM_SEGMENT(name)
+
+#include "tables/dmadata_table.h"
+
+#undef DEFINE_DMA_ENTRY
+#undef DEFINE_DMA_ENTRY_UNSET
+
+// dmadata Table definition
+#define DEFINE_DMA_ENTRY(name, _nameString) \
+    { SEGMENT_ROM_START(name), SEGMENT_ROM_END(name), SEGMENT_ROM_START(name), 0 },
+#define DEFINE_DMA_ENTRY_UNSET(name, _nameString) \
+    { SEGMENT_ROM_START(name), SEGMENT_ROM_END(name), 0xFFFFFFFF, 0xFFFFFFFF },
+
+DmaEntry gDmaDataTable[] = {
+#include "tables/dmadata_table.h"
+    { 0 },
```

Comment:
Is the `{ 0 },` here padding? Could it be made part of `sDmaDataPadding`?

Reply:
Nah. Afaik the last entry in the table needs to be all 0's (see `DmaMgr_FindDmaEntry` where it uses `entry->vromEnd == 0` to now it has reached the end)